lib/iris/coords.py

```diff
@@ -1792,20 +1789,10 @@ def bounds(self):
             (n_bounds, )``.
 
         """
-        if self._bounds is not None:
-            bounds = self._bounds
-            if is_lazy_data(bounds):
-                bounds = as_concrete_data(bounds,
-                                          nans_replacement=np.ma.masked)
-                # NOTE: we probably don't fully support for masked aux-coords.
-                # We certainly *don't* handle a _FillValue attribute (and
-                # possibly the loader will throw one away ?)
-                self._bounds = bounds
-            bounds = bounds.view()
+        if self._bounds_dm is not None:
+            return self._bounds_dm.data.view()
         else:
-            bounds = None
-
-        return bounds
+            return None
```

Comment:
I prefer the [approach that I've taken](https://github.com/dkillick/iris/blob/b9be68bb4174f609b0b7a28ad05ec18e8b915315/lib/iris/coords.py#L1691-L1701) that does not require multiple return statements. Multiple return statements can make code hard to follow and debug.